airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/record_filter.py

```diff
@@ -33,7 +33,12 @@ def filter_records(
         stream_slice: Optional[StreamSlice] = None,
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> Iterable[Mapping[str, Any]]:
-        kwargs = {"stream_state": stream_state, "stream_slice": stream_slice, "next_page_token": next_page_token}
+        kwargs = {
+            "stream_state": stream_state,
+            "stream_slice": stream_slice,
+            "next_page_token": next_page_token,
+            "stream_slice.extra_fields": stream_slice.extra_fields if stream_slice else {},
+        }
```

Comment:
can we add a test here that validates if we call `filter_records()` on the nested `stream_slice.extra_fields` context that we get the expected results

I only did a quick check so i might not have set up the interpolation right, but when I added a small test using `"{{ stream_slice.extra_fields.some_key }}"` and modified kwargs to look like this:
```
kwargs = {
            "stream_state": stream_state,
            "stream_slice": stream_slice,
            "next_page_token": next_page_token,
            "stream_slice.extra_fields": {"some_key": "some_value", "some_bool": True},
        }
```

I wasn't able to get this to resolve during the interpolation step to a real slice value. Once we add a unit test to confirm, then happy to 👍 

Do we need to nest the entire `extra_field` object under the `stream_slice` key in `kwargs`?

Reply:
The value can be accessed using `stream_slice.extra_fields["some_key"]`. I've added a test to validate this behaviour, and it has been tested during the regression tests for Jira.